### PR TITLE
Protractor SDK

### PR DESCRIPTION
**Pull request by kyryloonufriiev**
Add support for protractor SDK.
___
Also as a part of this task, I implemented reusable generic coverage tests code emitters. I think we can improve `TestSetup` abstraction. Also, I thought about "plain object presentation" for check settings, this will free us from the need to write the logic for building check settings several times. Instead, we will have a standard way to present check settings as a plain object, maybe it could be used not only for tests but for SDKs which doesn't have check settings abstraction, such as cypress.

### CODE REVIEW

File: packages/eyes-protractor/package.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    "protractor": ">=5.4.4"
+  },
+  "scripts": {
+    "lint": "eslint . --ext .js --no-eslintrc --config ../../.eslintrc",
+    "test": "yarn test:it",
+    "test:it": "mocha --timeout 120000 ./test/it/**/*.spec.js",
+    "coverage:generate": "coverage-tests create-tests --remote=$CVG_TESTS_REMOTE",
+    "coverage:sandbox": "yarn coverage:generate && yarn test:coverage && coverage-tests process-report",
+    "coverage:prod": "yarn coverage:generate && yarn test:coverage && coverage-tests process-report --send-report prod",
+    "test:coverage": "env APPLITOOLS_BATCH_NAME='JS Coverage Tests: eyes-protractor' APPLITOOLS_BATCH_ID=$(uuidgen) XUNIT_FILE=coverage-test-report.xml mocha-parallel-tests --no-timeouts --reporter spec-xunit-file --max-parallel 15 {./test/coverage/generic/*.spec.js,$(find ../sdk-coverage-tests/coverage-tests/custom/** -name '*.spec.js' -not -name '*.native.spec.js')}",
```
Comment (amitzur) [inline packages/eyes-protractor/package.json:71]: the `find` command didn't work on my linux machine. Can we use @tourdedave 's suggestion for grepping mocha?

**Comment by amitzur** [comment]
@kyryloonufriiev the generic coverage tests code emitting implementation is indeed what I was aiming for. After this PR we can consolidate all mocha code generation to one place.
cc @tourdedave

File: packages/eyes-protractor/test/coverage/index.js [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+    `const specs = require(path.resolve(process.cwd(), 'src/SpecWrappedDriver'))`,
+  )
+  result.storeHook(
+    'deps',
+    `const {
+      getEyes,
+      Browsers
+    } = require('@applitools/sdk-coverage-tests/coverage-tests/util/TestSetup')`,
+  )
+  result.storeHook('deps', `const {Target, RectangleSize, Region} = require('../../../index')`)
```
[the remark this answers is not in the document]
Comment (kyryloonufriiev) [reply packages/eyes-protractor/test/coverage/index.js:19]: We still can use it for all SDK's but I agree that `require(cwd)` is better.

File: packages/eyes-protractor/test/coverage/index.js [hunk 3]
```diff
@@ -0,0 +1,280 @@
+const supportedTests = require('./supported-tests')
+const {makeEmitTracker} = require('@applitools/sdk-coverage-tests')
+const sdkName = 'eyes-selenium'
+
+function initialize() {
+  const result = makeEmitTracker()
+  result.storeHook('deps', `const path = require('path')`)
```
[the remark this answers is not in the document]
Comment (kyryloonufriiev) [reply packages/eyes-protractor/test/coverage/index.js:7]: done

File: packages/eyes-protractor/test/coverage/index.js [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+            `await eyes.checkRegionBy(
+              ${makeRegionLocator(target)},
+              ${tag ? `'${tag}'` : undefined},
+              ${matchTimeout},
+              ${isFully}
+            )`,
+          )
+    } else {
+      result.storeCommand(`{`)
+      result.storeCommand(`const checkSettings = new Target()`)
```
[the remark this answers is not in the document]
Comment (kyryloonufriiev) [reply packages/eyes-protractor/test/coverage/index.js:134]: I'm sure we can do that if it is more comfortable for us. All in all `Target.window()` is just `new Target()`, `Target.region(...)` if `new Target().region(..)` and `Target.frame(...)` is `new Frame().frame(...)`

File: packages/eyes-protractor/package.json (hunk 1 above)
Comment (kyryloonufriiev) [reply packages/eyes-protractor/package.json:71]: done.

**Comment by kyryloonufriiev** [comment]
I added a test for element finding on the angular page. Also, I fixed `markScrollRootElement` in this PR.

**Pull request merged**